endpoints/openrtb2/video_auction.go

```diff
@@ -0,0 +1,495 @@
+	return jsonString, nil
+}
+
+func getVideoStoredRequestId(request []byte) (string, error) {
+	req := openrtb_ext.StoredRequestId{}
+
+	if err := json.Unmarshal(request, &req); err != nil {
+		return "", err
+	}
+	return req.StoredRequestId, nil
+
+}
+
+func mergeData(videoRequest *openrtb_ext.BidRequestVideo, bidRequest *openrtb.BidRequest) error {
+
+	if videoRequest.Site.Page != "" {
+		bidRequest.Site = &videoRequest.Site
+		if &videoRequest.Content != nil {
+			bidRequest.Site.Content = &videoRequest.Content
+		}
+	}
+
+	if videoRequest.App.Domain != "" {
+		bidRequest.App = &videoRequest.App
+		if &videoRequest.Content != nil {
+			bidRequest.App.Content = &videoRequest.Content
+		}
+	}
+
+	if &videoRequest.Device != nil {
+		bidRequest.Device = &videoRequest.Device
+	}
+
+	if &videoRequest.User != nil {
+		bidRequest.User = &openrtb.User{
+			BuyerUID: videoRequest.User.Buyeruids["appnexus"], //TODO: map to string merging
+			Yob:      videoRequest.User.Yob,
+			Gender:   videoRequest.User.Gender,
+			Keywords: videoRequest.User.Keywords,
+		}
+	}
+
+	bidExt, err := createBidExtension(videoRequest)
+	if err != nil {
+		return err
+	}
+	if len(bidExt) > 0 {
+		bidRequest.Ext = bidExt
+	}
+
+	bidRequest.Test = videoRequest.Test
+
+	bidRequest.TMax = 5000
+	return nil
+}
+
+func createBidExtension(videoRequest *openrtb_ext.BidRequestVideo) ([]byte, error) {
+
+	inclBrandCat := openrtb_ext.ExtIncludeBrandCategory{
+		PrimaryAdServer: videoRequest.IncludeBrandCategory.PrimaryAdserver,
+		Publisher:       videoRequest.IncludeBrandCategory.Publisher,
+	}
+	var durationRangeSec []int
+	if !videoRequest.PodConfig.RequireExactDuration {
+		durationRangeSec = videoRequest.PodConfig.DurationRangeSec
+	}
+
+	priceGranularity := openrtb_ext.PriceGranularityFromString("med")
+	if videoRequest.PriceGranularity.Precision != 0 {
+		priceGranularity = videoRequest.PriceGranularity
+	}
+
+	targeting := openrtb_ext.ExtRequestTargeting{
+		PriceGranularity:     priceGranularity,
+		IncludeWinners:       true,
+		IncludeBrandCategory: inclBrandCat,
+		DurationRangeSec:     durationRangeSec,
+	}
+
+	prebid := openrtb_ext.ExtRequestPrebid{
+		Targeting: &targeting,
+	}
+	extReq := openrtb_ext.ExtRequest{Prebid: prebid}
+
+	reqJSON, err := json.Marshal(extReq)
+	if err != nil {
+		return nil, err
+	}
+	return reqJSON, nil
+}
+
+func (deps *endpointDeps) parseVideoRequest(request []byte) (req *openrtb_ext.BidRequestVideo, errs []error) {
+	req = &openrtb_ext.BidRequestVideo{}
+
+	if err := json.Unmarshal(request, &req); err != nil {
+		errs = []error{err}
+		return
+	}
+
+	errL := deps.validateVideoRequest(req)
+	if len(errL) > 0 {
+		errs = append(errs, errL...)
+	}
+	return
+}
+
+func (deps *endpointDeps) validateVideoRequest(req *openrtb_ext.BidRequestVideo) []error {
+	errL := []error{}
+	if req.AccountId == "" {
+		err := errors.New("request missing required field: accountid")
+		errL = append(errL, err)
+	}
+	if req.StoredRequestId == "" {
```

Comment:
We wanted a config option if this should be required or not.

Reply:
Done